sdk/go/common/resource/plugin/langruntime_plugin.go

```diff
@@ -55,35 +56,125 @@ type langhost struct {
 // plugin could not be found, or an error occurs while creating the child process, an error is returned.
 func NewLanguageRuntime(host Host, ctx *Context, runtime, workingDirectory string, info ProgramInfo,
 ) (LanguageRuntime, error) {
-	path, err := workspace.GetPluginPath(ctx.Diag,
-		apitype.LanguagePlugin, strings.ReplaceAll(runtime, tokens.QNameDelimiter, "_"), nil, host.GetProjectPlugins())
+	attachPort, err := GetLanguageAttachPort(runtime)
 	if err != nil {
 		return nil, err
 	}
 
-	contract.Assertf(path != "", "unexpected empty path for language plugin %s", runtime)
+	var plug *plugin
+	var client pulumirpc.LanguageRuntimeClient
+	if attachPort != nil {
+		port := *attachPort
 
-	args, err := buildArgsForNewPlugin(host, info.RootDirectory(), info.Options())
-	if err != nil {
-		return nil, err
-	}
+		handshake := func(
+			ctx context.Context, bin string, prefix string, conn *grpc.ClientConn,
+		) (*pulumirpc.LanguageHandshakeResponse, error) {
+			req := &pulumirpc.LanguageHandshakeRequest{
+				EngineAddress:    host.ServerAddr(),
+				RootDirectory:    filepath.Dir(bin),
+				ProgramDirectory: filepath.Dir(bin),
+			}
+			return languageHandshake(ctx, bin, prefix, conn, req)
+		}
```

Comment:
Not sure what value the helper is adding here. Also, we pass a `req` only for it to be unpacked and reconstructed, which feels a bit off. Also I'm wondering if generally we should extend the `LanguageRuntime` interface in `sdk/go/common` in a similar way to how we've extended `Provider` -- so that this code and other pieces can operate at that level instead of on raw gRPC structs etc.

Reply:
So I did this to be consistent with the other code.  I don't really like it either but so far whenever I try to make something make more sense i break stuff and I've become a bit standoffish about deviating because I'm not sure I can trust our tests in this area to catch anything...I'm fine with doing what you suggest but I'd rather put it in another change. This is in fact exactly what provider_plugin.go does

Reply:
I think this is fine, it matches provider code and I can't think of a simpler way to do it quickly.